### PR TITLE
Frequency_vectorizer added

### PR DESCRIPTION
**Pull request by nassimgha**

### CODE REVIEW

File: src/text_processing_util_mds24/text_processing_util_mds24.py [hunk 1]
```diff
@@ -5,6 +5,27 @@ def text_clean(docs: list[str]) -> list[str]:
...
+
+    Parameters
+    ----------
+    docs : list[str]
+        A list of text documents.
+
+    Returns
+    -------
+    list[dict[str: float]]
+        A list of dictionaries, where each dictionary contains the word and the word's frequency in a text document.
```
Comment (jy1909) [inline src/text_processing_util_mds24/text_processing_util_mds24.py:19]: Small fix to be more specific.

```suggestion
        A list of dictionaries, where each dictionary contains each word and its frequency in a text document.
```

File: src/text_processing_util_mds24/text_processing_util_mds24.py [hunk 2]
```diff
@@ -5,6 +5,27 @@ def text_clean(docs: list[str]) -> list[str]:
...
+        
+    Examples
+    --------
+    >>> documents = ["This is a sample document.", "Another document for testing."]
+    >>> result = frequency_vectorizer(documents)
+    >>> print(result)
+    [{'This': 0.14, 'is': 0.14, 'a': 0.14, 'sample': 0.14, 'document.': 0.14},
+    {'Another': 0.16, 'document': 0.16, 'for': 0.16, 'testing.': 0.16}]
```
Comment (jy1909) [inline src/text_processing_util_mds24/text_processing_util_mds24.py:27]: I think there is a calculation issue here. Please review. Thanks.

```suggestion
    [{'This': 0.2, 'is': 0.2, 'a': 0.2, 'sample': 0.2, 'document.': 0.2},
    {'Another': 0.25, 'document': 0.25, 'for': 0.25, 'testing.': 0.25}]
```

**Review by jy1909 — CHANGES_REQUESTED** [review]
Overall looks fine. Just 2 small fixes. Thanks.

**Comment by nassimgha** [comment]
I fixed it and committed the changes. Thanks for your review and catching the mistakes.

**Comment by jy1909** [comment]
Okay, I see, thanks. Let's not merge our own PR and not commit directly on main in the future.

**Pull request merged**